pkg/controllers/nsxserviceaccount/nsxserviceaccount_controller.go

```diff
@@ -104,6 +176,12 @@ func (r *NSXServiceAccountReconciler) Reconcile(ctx context.Context, req ctrl.Re
 					return ResultRequeue, err
 				}
 			}
+			// update ProxyEndpoints if it has changed.
+			if err := r.Service.UpdateProxyEndpointsIfNeeded(ctx, obj); err != nil {
+				r.StatusUpdater.UpdateFail(ctx, obj, err, "", updateNSXServiceAccountStatuswithError)
+				return ResultRequeue, err
+			}
+			r.StatusUpdater.UpdateSuccess(ctx, obj, updateNSXServiceAccountStatus)
 			r.StatusUpdater.IncreaseDeleteSuccessTotal()
```

Comment:
Could you ask @yanjunz97 for this? I think this may be a mistake.

Reply:
I think we can remove `r.StatusUpdater.IncreaseDeleteSuccessTotal()` here. 
@yanjunz97 could you please take a look as well ?